src/RectangleB.java

```diff
@@ -0,0 +1,268 @@
+        return "Width=" + getWidth() + " Height=" + getHeight() + " PointSW=" + _pointSW.toString();
+    }
+
+    /**
+     * Calculates the perimeter of the rectangle
+     *
+     * @return The perimeter of the rectangle.
+     */
+    public int getPerimeter() {
+        return 2 * getWidth() + 2 * getHeight();
+    }
+
+    /**
+     * Calculates the area of the rectangle
+     *
+     * @return The area of the rectangle.
+     */
+    public int getArea() {
+        return getWidth() * getHeight();
+    }
+
+    /**
+     * Move the rectangle by deltaX in x direction and deltaY in y direction
+     *
+     * @param deltaX - translate the rectangle deltaX in the x direction.
+     * @param deltaY - translate the rectangle deltaY in the y direction.
+     */
+    public void move(int deltaX, int deltaY) {
+        _pointSW.move(deltaX, deltaY);
+        _pointNE.move(deltaX, deltaY);
+    }
+
+    /**
+     * Returns the width of the rectangle
+     *
+     * @return the width of the rectangle
+     */
+    public int getWidth() {
+        return _pointNE.getX() - _pointSW.getX();
+    }
+
+    /**
+     * Returns the height of the rectangle
+     *
+     * @return the height of the rectangle
+     */
+    public int getHeight() {
+        return _pointNE.getY() - _pointSW.getY();
+    }
+
+    /**
+     * Sets the width of the rectangle
+     *
+     * @param w - the width of the rectangle to set to
+     */
+    public void setWidth(int w) {
+        if (w >= MIN_WIDTH)
+            _pointNE.setX(_pointSW.getX() + w);
+    }
+
+    /**
+     * Sets the height of the rectangle
+     *
+     * @param h - the height of the rectangle to set to
+     */
+    public void setHeight(int h) {
+        if (h >= MIN_HEIGHT)
+            _pointNE.setY(_pointSW.getY() + h);
+    }
+
+    /**
+     * Returns true if the given rectangle is equal to this rectangle
+     *
+     * @param other - the rectangle to check equality with
+     * @return true if other and this rectangle are equal
+     */
+    public boolean equals(RectangleB other) {
+        return this._pointSW.equals(other._pointSW) &&
+                this._pointNE.equals(other._pointNE);
+    }
+
+    /**
+     * Returns the length of the rectangle diagonal
+     *
+     * @return the length of the diagonal of the Rectangle
+     */
+    public double getDiagonalLength() {
+        return _pointSW.distance(_pointNE);
+    }
+
+    /**
+     * Returns true if the current rectangle is larger than the parameter rectangle
+     *
+     * @param other - another Rectangle to compare with
+     * @return true - if the current Rectangle's area is larger than the other Rectangle which recieved as parameter, otherwise - false
+     */
+    public boolean isLarger(RectangleB other) {
+        return this.getArea() > other.getArea();
+    }
+
+    /**
+     * Changes the width to height and vice versa
+     */
+    public void changeSides() {
+        int temp = getWidth();
```

Comment:
Proposal:
verify that width is valid number for height, and that height is a valid number for height.
Only then, replace their values.

This requires some private method that will do validations for each of them!